bin/parse-legacy-file

```diff
@@ -1,4 +1,16 @@
 #!/usr/bin/env bash
-echo $(
-  sed '1 s/^v//' "$1"
-)
+
+version=$(sed '1 s/^v//' "$1")
+
+if [[ "$version" =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+  echo $version
+elif [[ "$version" = "node" ]]; then
+  echo $(
+    asdf list nodejs | tail -n1
+  )
+else
+  final_version=$(
+    asdf list nodejs | grep "^\s*$version" | tail -n1
+  )
+  [[ ! -z "$final_version" ]] && echo $final_version || echo $version
+fi
```

Comment:
I think we could move the translation logic to the install script, specifically into the [resolve_version_query](https://github.com/asdf-vm/asdf-nodejs/blob/5a46614ec24177146541543d3ec6327e136ae80f/bin/install#L94) function, and keep the parse-legacy-file simple

We would also remove the dependency of `asdf list` in this file, the `list` command is not meant to be used in quick scripts and perform a lot of operations that are not exactly light